### DISCUSSION TITLE
Powershell download fails behind proxy on windows: authentication not taken into account

### DISCUSSION THREAD

**Discussion by sylvlecl** [microsoft/vcpkg#38827] [Ideas]
**Describe the bug**

Same as  #8078 and #7207: 
behind our company proxy, on windows, the download of powershell timeouts.

The env variables HTTP_PROXY and HTTPS_PROXY are correctly set:
```cmd
set HTTP_PROXY=http://login:password@company-proxy.com:1234
...
```
Indeed, the download of dependencies, which is performed with curl does work (after having installed vcpkg and base tools like powershell outside from the company network).

**Some analysis**

After a quick look at vcpkg-tool source code and at [WinHTTP API](https://learn.microsoft.com/en-us/windows/win32/winhttp/authentication-in-winhttp), I suspect that the credentials part of the environment variables are not correctly provided to WinHTTP API.
Indeed, the API seems to expect the proxy URL, username, and password in separate options, whereas vcpkg provides the full content of HTTP_PROXY, including the credentials, as the proxy URL (see [this line](https://github.com/microsoft/vcpkg-tool/blob/cae13bd151c410aaa67829ff685ecb3adbf89408/src/vcpkg/base/downloads.cpp#L258)).

Indeed, when using windows provided command line tool to set those winhttp options, a URL with credentials is considered as "invalid  parameter", which seems to confirm that winhttp expects a plain URL here:
```cmd
>netsh winhttp set proxy http://toto:tt@test.com
Erreur lors de l'écriture des paramètres proxy. (87) Paramètre incorrect.
```

**Environment**
- OS: Windows

**To Reproduce**

Behind a proxy requiring some credentials:
```cmd
set HTTP_PROXY=http://login:password@company-proxy.com:1234
set HTTPS_PROXY=%HTTP_PROXY%
# Try install any port
vcpkg install jsoncpp
```

**Expected behavior**
I would expect the use of HTTP*_PROXY environment variables to be consistent between the downloads that use WinHTTP API and the downloads which use CURL, so that I just need to set those environment variables to have the whole chain working, like on Linux.

**Comment by j-xella** [comment]
As a workaround, I currently build and run an instance of cntlm proxy server 🤦‍♂️🤦‍♂️🤦‍♂️

Generally speaking, for a product created by microsoft, endorsed by microsoft and now being a part of Visual Studio, it is hard to comprehend how corporate users are neglected. It seems that this project is geared toward enthusiasts doing development on their home PC, where you have direct internet access without any firewalls and can install anything you want on your PC. Complete opposite of the corporate world, where you are under proxy firewall that does SSL interception of all the traffic, and have no admin rights on you PC whatsoever. Then every aspect of using vcpkg is a pain. You see multitude youtube videos where libraries are being added in seconds with `vcpkg install xxx`, and your experience is just opposite of that.

**Comment by saperlot** [comment]
same issues here, i am also behind a corporate proxy with login.
